modules/core/src/v2/coins/eos.ts

```diff
@@ -408,4 +441,315 @@ export class Eos extends BaseCoin {
       .call(this)
       .asCallback(callback);
   }
+
+  /**
+   * Validate a public or private key
+   * If passphrase is provided, try to decrypt the key with it
+   * @param key
+   * @param source
+   * @param passphrase
+   * @param isUnsignedSweep
+   * @param isKrsRecovery
+   */
+  validateKey({ key, source, passphrase, isUnsignedSweep, isKrsRecovery }: ValidateKeyOptions): HDNode {
+    if (!key.startsWith('xprv') && !isUnsignedSweep) {
+      // Try to decrypt the key
+      try {
+        if (source === 'user') {
+          return HDNode.fromBase58(sjcl.decrypt(passphrase, key));
+        }
+        if (source === 'backup' && !isKrsRecovery) {
+          return HDNode.fromBase58(sjcl.decrypt(passphrase, key));
+        }
+      } catch (e) {
+        throw new Error(`Failed to decrypt ${source} key with passcode - try again!`);
+      }
+    }
+    try {
+      return HDNode.fromBase58(key);
+    } catch (e) {
+      throw new Error(`Failed to validate ${source} key - try again!`);
+    }
+  }
+
+  /**
+   * Prepare and validate all keychains from the keycard for recovery
+   * @param userKey
+   * @param backupKey
+   * @param recoveryDestination
+   * @param krsProvider
+   * @param walletPassphrase
+   */
+  initiateRecovery({
+    userKey,
+    backupKey,
+    recoveryDestination,
+    krsProvider,
+    walletPassphrase,
+  }: BaseCoinInitiateRecoveryOptions): Bluebird<HDNode[]> {
+    return co(function*() {
+      const isKrsRecovery = backupKey.startsWith('xpub') && !userKey.startsWith('xpub');
+      const isUnsignedSweep = backupKey.startsWith('xpub') && userKey.startsWith('xpub');
+
+      if (isKrsRecovery) {
+        const krsProviderConfig = config.krsProviders[krsProvider];
+        if (_.isUndefined(krsProviderConfig)) {
+          throw new Error('unknown key recovery service provider');
+        }
+
+        if (!krsProviderConfig.supportedCoins.includes(this.getFamily())) {
+          throw new Error('specified key recovery service does not support recoveries for this coin');
+        }
+      }
+
+      const keys = [
+        this.validateKey({
+          key: userKey,
+          source: 'user',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+        this.validateKey({
+          key: backupKey,
+          source: 'backup',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+      ];
+      if (!this.isValidAddress(recoveryDestination)) {
+        throw new Error('Invalid destination address!');
+      }
+
+      return keys;
+    }).call(this);
+  }
+
+  /**
+   * Make a request to one of the public EOS nodes available
+   * @param endpoint
+   * @param payload
+   */
+  getDataFromNode({ endpoint, payload }: { endpoint: string; payload?: object }): Bluebird<any> {
```

Comment:
Can this be made protected? Or does this need to be part of the public API?

Same for other helper methods below - I'd rather not expose these unless really needed.